File: src/lessons/welcome/SlugTrackingEntity.py
import java.awt.Color as Color

# BEGIN TEMPLATE
def isFacingTrail(trailColor):
   # write your code
   # BEGIN SOLUTION
   if isFacingWall():
      return False
   else:
	  forward()
	  res = getGroundColor().toString() == trailColor.toString()
	  backward()
	  return res
# END TEMPLATE

def hunt():
   while not isOverBaggle():
      brushUp()
      if isFacingTrail(Color.green):
         brushDown()
         forward()
         brushUp()
      else:
	     turnLeft()
   pickUpBaggle()

hunt()

File: src/lessons/welcome/methods/slug/SlugTrackingEntity.scala
package lessons.welcome.methods.slug;

import java.awt.Color;

class SlugTrackingEntity extends plm.universe.bugglequest.SimpleBuggle {

	override def run() {
		while (! isOverBaggle()) {
			if (isFacingTrail()) {
				brushDown();
				stepForward();
				brushUp();
			} else {
				left();
			}
		}
		pickupBaggle();
	}

	/* BEGIN TEMPLATE */
	def isFacingTrail():Boolean = {
		// Write your code here
		/* BEGIN SOLUTION */
		if (isFacingWall())
			return false;
		stepForward();
		val res = (getGroundColor() == Color.green); 
		stepBackward();
		return res;
		/* END SOLUTION */
	}		
	/* END TEMPLATE */

}
